packages/astro-ghostcms/src/astro-ghostcms.ts

```diff
@@ -0,0 +1,315 @@
+					GhostENVLogger.error(
+						c.bgRed(
+							c.bold(
+								c.white("CONTENT_API_KEY is not set in environment variables"),
+							),
+						),
+					);
+					throw new AstroError(
+						`${name} CONTENT_API_KEY is not set in environment variables`,
+					);
+				}
+
+				if (options.ghostURL === undefined) {
+					GhostENVLogger.warn(
+						c.bgYellow(
+							c.bold(
+								c.black(
+									"ghostURL is not set in user configuration falling back to environment variable",
+								),
+							),
+						),
+					);
+					if (ENV.CONTENT_API_URL === undefined) {
+						GhostENVLogger.error(
+							c.bgRed(
+								c.bold(
+									c.white(
+										"CONTENT_API_URL is not set in environment variables",
+									),
+								),
+							),
+						);
+						throw new AstroError(
+							`${name} CONTENT_API_URL is not set in environment variables`,
+						);
+					}
+				}
+				GhostENVLogger.info(
+					c.bold(c.green("GhostCMS environment variables are set")),
+				);
+
+				// Set up Astro-GhostCMS Integrations
+				GhostIntegrationLogger.info(
+					c.bold(c.magenta("Configuring Enabled Integrations")),
+				);
+
+				// Theme Provider
+				if (!options.ThemeProvider?.disableThemeProvider) {
+					addIntegration(
+						ghostThemeProvider({
+							theme: options.ThemeProvider?.theme,
+							verbose,
+						}),
+					);
+				} else if (verbose) {
+						GhostIntegrationLogger.info(c.gray("Theme Provider is disabled"));
+				}
+
+				// Satori OG Images
+				if (options.enableOGImages) {
+					addIntegration(ghostOGImages({ verbose }));
+				} else if (verbose) {
+						GhostIntegrationLogger.info(
+							c.gray("OG Image Provider is disabled"),
+						);
+				}
+
+				// RSS Feed
+				if (options.enableRSSFeed) {
+					addIntegration(ghostRSS({ verbose }));
+				} else if (verbose) {
+						GhostIntegrationLogger.info(c.gray("RSS Feed is disabled"));
+				}
+
+				// @ASTROJS/SITEMAP
+				if (!hasIntegration("@astrojs/sitemap")) {
+					if (verbose) {
+						GhostIntegrationLogger.info(
+							c.bold(
+								c.magenta(`Adding ${c.blue("@astrojs/sitemap")} integration`),
+							),
+						);
+					}
+					addIntegration(sitemap(options.Integrations?.sitemap));
+				} else if (verbose) {
+						GhostIntegrationLogger.info(
+							c.gray(
+								"@astrojs/sitemap integration already exists, skipping...",
+							),
+						);
+				}
+				// ASTRO-ROBOTS-TXT
+				if (!hasIntegration("astro-robots-txt")) {
+					if (verbose) {
+						GhostIntegrationLogger.info(
+							c.bold(
+								c.magenta(`Adding ${c.blue("astro-robots-txt")} integration`),
+							),
+						);
+					}
+					addIntegration(robotsTxt(options.Integrations?.robotsTxt));
+				} else if (verbose) {
+						GhostIntegrationLogger.info(
+							c.gray(
+								"astro-robots-txt integration already exists, skipping...",
+							),
+						);
+				}
+
+				// Set up default 404 page
+				if (!options.disableDefault404) {
+					if (verbose) {
```

Comment:
Instead of all these manual checks for `verbose` you'd probably be better off with a util function which checks it instead.

```ts
const log = message => {
  if (verbose) {
    console.log(...)
  }
}
```

Or could be a Florian special and return a console log function:

```ts
const createLogger = verbose => message => {
  if (verbose) {
    console.log(...)
  }
}

const log = createLogger(options.verbose)

log(whatever) // now you don't have to check everywhere
```